### ISSUE TITLE
[Research] Add Whisper-Based Multi-Condition Mental Health Screening (Depression/Anxiety/Insomnia/Fatigue)

### ISSUE THREAD

**Issue by LakshmiSravyaVedantham** [LakshmiSravyaVedantham/DreamAnalysis#41]
## What

Build a multi-condition mental health screening pipeline using **Whisper encoder embeddings** as the backbone. JMIR 2024 study (865 adults) achieves:
- Depression: **AUC 0.76-0.78**
- Anxiety: **AUC 0.77**  
- Insomnia: **AUC 0.73**
- Fatigue: **AUC 0.68**

All from a single speech sample analyzed with Whisper Medium + max pooling.

## Why

- NeuralDreamWorkshop monitors sleep (staging model), stress (EEG), and emotions — but has no dedicated depression/anxiety/insomnia screening
- Voice is a passive, low-burden way to screen (no questionnaire fatigue)
- Whisper Medium embeddings capture prosodic + phonetic + speaker characteristics in one representation
- Combined with PHQ-9/GAD-7 (issue #25) as ground truth, this creates a validated screening system
- Insomnia detection from voice complements the sleep staging model

### Step 1: Voice Activity Detection + Feature Extraction
```python
# ml/models/voice_mental_health.py

from silero_vad import get_speech_timestamps, read_audio
import whisper

class VoiceMentalHealthScreener:
    """Multi-condition mental health screening from speech.
    
    Pipeline: VAD → Whisper encoder → max pooling → binary classifiers
    Conditions: depression, anxiety, insomnia, fatigue
    Minimum audio: 30 seconds of speech
    """
    
    def __init__(self, whisper_size="small"):  # small=244M, good balance
        self.whisper_model = whisper.load_model(whisper_size)
        self.classifiers = {
            "depression": load_classifier("depression"),
            "anxiety": load_classifier("anxiety"),
            "insomnia": load_classifier("insomnia"),
            "fatigue": load_classifier("fatigue"),
        }
    
    def extract_embeddings(self, audio):
        """Extract Whisper encoder embeddings with max pooling."""
        mel = whisper.log_mel_spectrogram(audio).to(self.device)
        with torch.no_grad():
            encoder_output = self.whisper_model.encoder(mel.unsqueeze(0))
        # Max pooling over time dimension
        pooled = encoder_output.max(dim=1).values  # (1, embed_dim)
        return pooled.cpu().numpy()
    
    def screen(self, audio_path) -> dict:
        """Screen for multiple conditions.
        
        Returns:
            {
                "depression": {"risk_score": 0.0-1.0, "severity": "...", "confidence": 0.0-1.0},
                "anxiety": {"risk_score": 0.0-1.0, ...},
                "insomnia": {"risk_score": 0.0-1.0, ...},
                "fatigue": {"risk_score": 0.0-1.0, ...},
                "model": "whisper-small",
                "audio_duration_sec": 45.2
            }
        """
```

### Step 2: Training pipeline
- Use DAIC-WOZ dataset (depression/anxiety interviews, 189 participants)
- Split Whisper embeddings → train binary SVM/MLP per condition
- Add uncertainty estimation via Monte Carlo dropout
- Oversample minority class (depressed/anxious)

### Step 3: API endpoint
```
POST /voice/mental-health-screen
  Input: audio file (30-60s minimum)
  Output: risk scores for depression, anxiety, insomnia, fatigue
  
GET /voice/mental-health-history/{user_id}
  Output: trend of risk scores over time
```

### Step 4: Combine with EEG
- Fuse voice risk scores with EEG stress index + FAA valence
- Multimodal depression indicator: voice (AUC 0.77) + EEG FAA (adds +5-10%)
- Show on study profile alongside PHQ-9/GAD-7 scores

### Files to Modify
- `ml/models/voice_mental_health.py` (NEW) — Multi-condition screener
- `ml/training/train_voice_mental_health.py` (NEW) — Training script
- `ml/api/routes.py` — Add screening endpoints
- `client/src/pages/study-profile.tsx` — Show mental health risk trends

### Expected Impact
- New capability: passive depression/anxiety/insomnia/fatigue screening
- Depression detection: **AUC 0.76-0.78** from voice alone
- Combined with EEG: **AUC 0.82-0.85** (estimated)

## Complexity
MEDIUM-HIGH — Whisper Small (244M params) needs GPU or quantization for real-time

## References
- JMIR 2024 multi-condition: https://www.jmir.org/2024/1/e58572
- Kintsugi depression (14K participants): https://pubmed.ncbi.nlm.nih.gov/39805690/
- StressSpeak personalization (MDPI 2025): https://pmc.ncbi.nlm.nih.gov/articles/PMC12651002/
